app/controllers/ems_common.rb

```diff
@@ -319,18 +319,24 @@ def update_button_validate
   end
   private :update_button_validate
 
-  def validate_credentials(verify_ems)
+  def get_validation_object(verify_ems)
```

Comment:
I was thinking as well about moving it to separate module, but this method uses `set_record_vars` method as well. I am not sure whether it's good to move just these two methods to new module and keep set_record_vars where it is. If so, `get_validation_object` method would require to have method `set_record_vars` in scope. But if that's normal ruby development I can move it (`get_validation_object` and `validate_credentials`) to different module.